source/adios2/core/ADIOS.cpp

```diff
@@ -79,6 +79,18 @@ IO &ADIOS::DeclareIO(const std::string ioName)
     return ioPair.first->second;
 }
 
+IO &ADIOS::GetIO(const std::string name)
+{
+    auto itIO = m_IOs.find(name);
+    if (itIO == m_IOs.end())
```

Comment:
It doesn't throw anything.  Dereferencing an end() iterator is undefined and is effectively an access violation.